scripts/linters/pylint_extensions_test.py

```diff
@@ -2234,3 +2234,191 @@ def test_file_overview_at_end_of_file(self):
+        self.checker_test_object.setup_method()
+
+    def test_no_newline_below_class_docstring(self):
+        node_no_newline_below_class_docstring = astroid.scoped_nodes.Module(
+            name='test',
+            doc='Custom test')
+        temp_file = tempfile.NamedTemporaryFile()
+        filename = temp_file.name
+
+        with python_utils.open_file(filename, 'w') as tmp:
+            tmp.write(
+                u"""
+                    class ClassName(Class):
+                        \"\"\"This is a docstring.\"\"\"
+                        a = 1 + 2
+                """)
+        node_no_newline_below_class_docstring.file = filename
+        node_no_newline_below_class_docstring.path = filename
+
+        self.checker_test_object.checker.process_module(
+            node_no_newline_below_class_docstring)
+
+        message = testutils.Message(
+            msg_id='newline-below-class-docstring',
+            line=4)
+
+        with self.checker_test_object.assertAddsMessages(message):
+            temp_file.close()
+
+    def test_excessive_newline_below_class_docstring(self):
+        node_excessive_newline_below_class_docstring = (
+            astroid.scoped_nodes.Module(
+                name='test',
+                doc='Custom test'))
+        temp_file = tempfile.NamedTemporaryFile()
+        filename = temp_file.name
+
+        with python_utils.open_file(filename, 'w') as tmp:
+            tmp.write(
+                u"""
+                    class ClassName(Class):
+                        \"\"\"This is a docstring.\"\"\"
+
+
+                        a = 1 + 2
+                """)
+        node_excessive_newline_below_class_docstring.file = filename
+        node_excessive_newline_below_class_docstring.path = filename
+
+        self.checker_test_object.checker.process_module(
+            node_excessive_newline_below_class_docstring)
+
+        message = testutils.Message(
+            msg_id='newline-below-class-docstring',
+            line=6)
+
+        with self.checker_test_object.assertAddsMessages(message):
+            temp_file.close()
+
+    def test_class_with_no_docstring(self):
+        node_class_with_no_docstring = (
+            astroid.scoped_nodes.Module(
+                name='test',
+                doc='Custom test'))
+        temp_file = tempfile.NamedTemporaryFile()
+        filename = temp_file.name
+
+        with python_utils.open_file(filename, 'w') as tmp:
+            tmp.write(
+                u"""
+                    class ClassName(Class):
+                        def func():
+                            \"\"\"This is a docstring.\"\"\"
+                            a = b + c
+                """)
+        node_class_with_no_docstring.file = filename
+        node_class_with_no_docstring.path = filename
+
+        self.checker_test_object.checker.process_module(
+            node_class_with_no_docstring)
+
+        with self.checker_test_object.assertNoMessages():
```

Comment:
Don't we expect to have a newline before the function in this case?

Reply:
No, because this class has no docstring.

Reply:
Still, we need a newline after the class, right? Like the way you have written the setUp function for this test class.

Reply:
All classes have docstring except few Tests classes, I think we have lint check to have docstring for all classes in the future. That will solve this problem.

Reply:
My question still persists do we expect a newline after class defination even if the class deosn't have the docstring?

Reply:
My question still persists do we expect a newline after class defination even if the class deosn't have the docstring?

Reply:
No, we just need a newline after the class docstring.